modules/300-prometheus/hooks/alertmanager_crd_discovery.go

```diff
@@ -270,37 +198,17 @@ func parseTargetCR(am Alertmanager) (alertmanagerValue, error) {
 	return value, nil
 }
 
-func applyDeprecatedAlertmanagerServiceFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
-	service := &corev1.Service{}
-	err := sdk.FromUnstructured(obj, service)
-	if err != nil {
+func parseInternalCR(am Alertmanager) (alertmanagerInternal, error) {
+	value, ok, err := unstructured.NestedMap(am.Spec, "internal")
+	if err != nil || !ok {
```

Comment:
Could there be a situation where err is nil, but ok is false? We will return nil, which is, I assume, not expected behavior.

Reply:
Yes, good point. Fixed, thanks.